File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i, a in enumerate(nums):
            sum = target - a
            if sum in dict:
                return [i, dict[sum]]
            else:
                dict[a] = i
                

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<>();

        for(int i = 0; i < nums.length; i++) {
            int sum = target - nums[i];

            if(hm.containsKey(sum)) {
                return new int[] {hm.get(sum), i};
            }
            else {
                hm.put(nums[i], i);
            }
        }

        return new int[] {};
    }
}